ui/packages/consul-ui/app/routes/settings.js

```diff
@@ -1,33 +1,50 @@
-import { inject as service } from '@ember/service';
 import Route from 'consul-ui/routing/route';
-import { hash } from 'rsvp';
-import { get, set, action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { get, action } from '@ember/object';
 
 export default class SettingsRoute extends Route {
-  @service('client/http')
-  client;
-
-  @service('settings')
-  repo;
-
-  @service('repository/dc')
-  dcRepo;
-
-  @service('repository/nspace/disabled')
-  nspacesRepo;
-
-  model(params) {
-    const app = this.modelFor('application');
-    return hash({
-      item: this.repo.findAll(),
-      dc: this.dcRepo.getActive(undefined, app.dcs),
-      nspace: this.nspacesRepo.getActive(),
-    }).then(model => {
-      if (typeof get(model.item, 'client.blocking') === 'undefined') {
-        set(model, 'item.client', { blocking: true });
-      }
-      return model;
+  @service('client/http') client;
+
+  @service('settings') repo;
+  @service('repository/dc') dcRepo;
+  @service('repository/permission') permissionsRepo;
+  @service('repository/nspace/disabled') nspacesRepo;
+
+  async model(params) {
+    // reach up and grab things from the application route/controller
+    const app = this.controllerFor('application');
```

Comment:
I think all this stuff could all be done cleaner now we have our Routlets, but I didn't want to change too much unrelated stuff in this PR, so I left that for the future.